update_list.py

```diff
@@ -0,0 +1,351 @@
+        if len(self.ua_list) == 0:
+            return
+
+        if self.append:
+            with open(full_path, "a") as writer:
+                writer.write("\n".join(ua_list))
+        else:
+            with open(full_path, "w") as writer:
+                writer.write("\n".join(ua_list))
+
+    @staticmethod
+    def _get_platform(user_agent: UserAgent):
+        """Get the platform as a string."""
+        if user_agent.is_mobile:
+            return "mobile"
+        elif user_agent.is_tablet:
+            return "tablet"
+        elif user_agent.is_pc:
+            return "pc"
+        elif user_agent.is_bot:
+            return "bot"
+        else:
+            return "other"
+
+    def _get_browser(self, user_agent: UserAgent) -> str:
+        """Get the browser as a string."""
+        browser = user_agent.browser.family
+
+        if re.search(self.chrome_pattern, browser):
+            return "chrome"
+        elif re.search(self.firefox_pattern, browser):
+            return "firefox"
+        elif re.search(self.safari_pattern, browser):
+            return "safari"
+        elif re.search(self.edge_pattern, browser):
+            return "edge"
+
+        return browser
+
+    def _get_os(self, user_agent: UserAgent) -> str:
+        """Get the operating system as a string."""
+        os = user_agent.os.family
+        if re.search(self.ios_pattern, os):
+            return "ios"
+        elif re.search(self.linux_pattern, os):
+            return "linux"
+        elif re.search(self.macos_pattern, os):
+            return "macos"
+        elif re.search(self.windows_pattern, os):
+            return "win10"
+        elif re.search(self.android_pattern, os):
+            return "android"
+
+        return os
+
+    @staticmethod
+    def _get_version(user_agent: UserAgent) -> float:
+        "Get the browser version as a float."
+        version_groups = user_agent.browser.version_string.split(".")
+        if len(version_groups) == 0 or version_groups[0] == "":
+            return 0.0
+        if len(version_groups) == 1:
+            return float(version_groups[0])
+        else:
+            return float(".".join(version_groups[:2]))
+
+
+if __name__ == "__main__":
+    updater = UserAgentUpdater(".")
+
+    # Parse the command line arguments, if provided.
+    parser = argparse.ArgumentParser(
+        description="Script to update the list of user agents."
+    )
+    parser.add_argument(
+        "browser_args",
+        nargs="*",
+        default=[],
+        help="Arbitrary number of browsers to include in the update",
+    )
+    parser.add_argument(
+        "--limit",
+        default=200,
+        type=int,
+        help="The max number of user agents per browser",
+    )
+    parser.add_argument(
+        "--max_version_lag",
+        type=float,
+        nargs="*",
+        default=0.0,
+    )
+
+    args = parser.parse_args()
+    possible_browsers = [
```

Comment:
Maybe make `possible_browsers` also argument input (ArgumentParser) on the script?

When running this for the first time.. I directly get a 429 error (too many requests). 

```sh
  File "/media/melroy/Data/Projects/fake-useragent/./update_list.py", line 347, in <module>
    new_firefox_useragents = updater.get_user_agents(
                             ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/media/melroy/Data/Projects/fake-useragent/./update_list.py", line 106, in get_user_agents
    user_agents = self.send_user_agent_request(
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/media/melroy/Data/Projects/fake-useragent/./update_list.py", line 151, in send_user_agent_request
    user_agents = self.send_post_request(
                  ^^^^^^^^^^^^^^^^^^^^^^^
  File "/media/melroy/Data/Projects/fake-useragent/./update_list.py", line 75, in send_post_request
    response.raise_for_status()
  File "/home/melroy/.venvs/fake_user_agent_testing_env/lib/python3.12/site-packages/requests/models.py", line 1024, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 429 Client Error: Too Many Requests for url: https://user-agents.net/download
```
So my default maybe only try one browser at the time (by default)? Or I get blocked by Cloudflare is also possible.